### DISCUSSION TITLE
Missing pre-change data for device bay

### DISCUSSION THREAD

**Discussion by blipnet** [netbox-community/netbox#11441] [General]
NetBox Version: v3.3.10
Python Version: 3.8.6

When updating device bays by adding or removing a device from the bay the pre-change data is missing.

In WebGUI there is following warning: Warning: Comparing non-atomic change to previous change record
REST API: prechange_data is null
Webhook: snapshot -> prechange is null

There is same behavior both when adding or removing a device.

I'm working with webhooks to update our asset management system, when someone change the device in a device bay this needs to be updated.

So when someone removes a device from device bay the webhook snapshot is (it's the same in REST API):
    "snapshots": {
        "prechange": null,
        "postchange": {
            "created": "2021-06-29T00:00:00Z",
            "last_updated": "2023-01-05T11:39:15.498Z",
            "device": 4761,
            "name": "6",
            "label": "Blade 6",
            "description": "",
            "installed_device": null,
            "custom_fields": {},
            "tags": []
        }
    }

Shouldn't the previous installed device be in prechange?
I do get this behavior when installing a device in device bay but not when removing.

I don't know if this is a bug or if I don't understand how to retrieve the data

**Comment by kkthxbye-code** [comment]
That's a bug, DeviceBayPopulateView and DeviceBayDepopulateView doesn't call snapshot before disassociating the device. Please create an issue with the required replication steps.

**Reply by blipnet** [reply]
Thanks for quick feedback
